src/target/nixos.rs

```diff
@@ -0,0 +1,210 @@
+        default::file_exists(path)
+    }
+
+    #[allow(unused_variables)]
+    fn file_delete(host: &mut Host, path: P) -> Result<()> {
+        default::file_delete(path)
+    }
+
+    #[allow(unused_variables)]
+    fn file_mv(host: &mut Host, path: P, new_path: P) -> Result<()> {
+        default::file_mv(path, new_path)
+    }
+
+    #[allow(unused_variables)]
+    fn file_copy(host: &mut Host, path: P, new_path: P) -> Result<()> {
+        default::file_copy(path, new_path)
+    }
+
+    #[allow(unused_variables)]
+    fn file_get_owner(host: &mut Host, path: P) -> Result<FileOwner> {
+        linux::file_get_owner(path)
+    }
+
+    #[allow(unused_variables)]
+    fn file_set_owner(host: &mut Host, path: P, user: &str, group: &str) -> Result<()> {
+        default::file_set_owner(path, user, group)
+    }
+
+    #[allow(unused_variables)]
+    fn file_get_mode(host: &mut Host, path: P) -> Result<u16> {
+        linux::file_get_mode(path)
+    }
+
+    #[allow(unused_variables)]
+    fn file_set_mode(host: &mut Host, path: P, mode: u16) -> Result<()> {
+        default::file_set_mode(path, mode)
+    }
+}
+
+//
+// Package
+//
+
+impl PackageTarget for NixOsTarget {
+    fn default_provider(host: &mut Host) -> Result<Providers> {
+        default::default_provider(host, vec![Providers::Nix])
+    }
+}
+
+//
+// Service
+//
+
+impl ServiceTarget for NixOsTarget {
+    #[allow(unused_variables)]
+    fn service_action(host: &mut Host, name: &str, action: &str) -> Result<Option<CommandResult>> {
+        linux::service_systemd(name, action)
+    }
+}
+
+//
+// Telemetry
+//
+
+impl TelemetryTarget for NixOsTarget {
+    #[allow(unused_variables)]
+    fn telemetry_init(host: &mut Host) -> Result<Value> {
+        let cpu_vendor = try!(linux::cpu_vendor());
+        let cpu_brand = try!(linux::cpu_brand_string());
+        let hostname = try!(default::hostname());
+        let (version, maj, min, patch) = try!(version());
+
+        let telemetry = Telemetry::new(
+            Cpu::new(
+                &cpu_vendor,
+                &cpu_brand,
+                try!(linux::cpu_cores()),
+            ),
+            try!(default::fs()),
+            &hostname,
+            try!(linux::memory()),
+            try!(linux::net()),
+            Os::new(
+                env::consts::ARCH,
+                "linux",
+                "nixos",
+                &version,
+                maj,
+                min,
+                patch
+            ),
+        );
+
+        Ok(telemetry.into_value())
+    }
+}
+
+fn version() -> Result<(String, u32, u32, u32)> {
+    let out = process::Command::new("nixos-version").output()?;
+    let version_str = str::from_utf8(&out.stdout).or(Err(Error::Generic("Could not read OS version".into())))?.trim();
+    let mut parts = version_str.split('.');
+    let version_maj = parts.next().ok_or(Error::Generic(format!("Expected OS version format `u32.u32`. Got: {}", version_str)))?.parse()?;
```

Comment:
Hrm...that's the wrong code sample. Anyway, see the bottom of `src/target/nixos.rs` for those `expect()` lines.